torch/csrc/jit/type.h

```diff
@@ -137,6 +137,13 @@ struct TensorType : public Type {
     t->strides_ = TensorType::contiguousStridesOf(sizes_);
     return t;
   }
+
+  TypePtr toScalarType(at::ScalarType type){
```

Comment:
Can we add `TensorTypePtr` and return it from here? It's not an inherited method, so it's ok to change the return type, and it will avoid us unnecessary `expect<TensorType>` everywhere else.

Reply:
I was wondering why `contiguous()` returns TypePtr and not TensorTypePtr. If you think it's ok, I'll make both `contiguous` and `toScalarType` return TensorTypePtr.

Reply:
Sure, I think that makes sense.